interface/forms/LBF/new.php

```diff
@@ -7,12 +7,28 @@
  * @link      http://www.open-emr.org
  * @author    Rod Roark <rod@sunsetsystems.com>
  * @author    Brady Miller <brady.g.miller@gmail.com>
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
  * @copyright Copyright (c) 2009-2019 Rod Roark <rod@sunsetsystems.com>
  * @copyright Copyright (c) 2019 Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (c) 2018-2020 Jerry Padgett <sjpadgett@gmail.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
 
-require_once(__DIR__ . "/../../globals.php");
+if (isset($_GET['isPortal']) && (int)$_GET['isPortal'] !== 0) {
+    require_once(__DIR__ . "/../../../src/Common/Session/SessionUtil.php");
+    OpenEMR\Common\Session\SessionUtil::portalSessionStart();
+    if (isset($_SESSION['pid']) && isset($_SESSION['patient_portal_onsite_two'])) {
+        $ignoreAuth_onsite_portal_two = true;
+        $ignoreAuth = true;
+    } else {
+        $ignoreAuth_onsite_portal_two = false;
+        $ignoreAuth = false;
+    }
+    // do not destroy portal session.
+    // @todo add redirect portal landing on fail.
+}
+
+require_once("../../globals.php");
```

Comment:
Wouldn't this be more straightforward to figure out if being called via portal versus core:
https://github.com/openemr/openemr/blob/master/library/ajax/easipro_util.php#L15-L30

Reply:
Also, I haven't checked out the code in details yet, but are we making sure that stuff being worked on is binded to the pid in $_SESSION['pid'] (ie. can't look at or submit other patient data from the patient portal)?

Reply:
btw, there are some similar blocks (scripts that support both portal and core) that have a redirect in it (can find these by searching code for `portalSessionCookieDestroy`)

Reply:
nope, I want to ensure that a call is specific from portal(an iframe in this case) not just that a portal session exist.

Reply:
Sounds good.
Can remove the 2 `$ignoreAuth` lines above (I actually forgot what `$ignoreAuth_onsite_portal_two` did and just reviewed the code; turns out is basically sets ignoreAuth to true if the portal is on, so that is why no need for the ignoreAuth flags also).
Would consider the following line below the else:
```
OpenEMR\Common\Session\SessionUtil::portalSessionCookieDestroy();
```
(this is basically just clean up to avoid a bunch of sessions piling up)

Reply:
For pid, you shouldn't be able set to a pid in new.php because workflow doesn't rely on it. Lookups, if needed, are by lbf data records tied to a pid set in portal and stored in onsite documents. Remembering also that I can't be tied to a session pid because review requires I access various pids.

Reply:
Actually, turns out the `OpenEMR\Common\Session\SessionUtil::portalSessionCookieDestroy();` is mandatory after the else (if the portal session is not valid) or else it will break the core session loading in globals.php (https://github.com/openemr/openemr/blob/master/interface/globals.php#L113) and could potentially then lead to a security issue.

Reply:
Okay thanks, I was wondering early on about some session issues. This may be it. Still in this case we won't fall through to get to the globals check because this is a direct portal request where either I can proceed or not. But, a cleanup can't hurt.

Also I think I won't redirect but fail silent. The portal will cleanup if user is in an authorized login and request fails for one reason or another where, with outside attacks, I was always taught to never give any information on a security fail. A redirect says, try again(hacker may not even know the url for portal login). Silence says many other things that is hard to pinpoint.